mlir-tensorrt/python/bindings/Runtime/RuntimePyBind.cpp

```diff
@@ -119,6 +120,16 @@ struct type_caster<MTRT_MemRefValue> {
 
 namespace {
 
+bool isGPUPointer(void *ptr) {
+  cudaPointerAttributes attributes;
+  cudaError_t status = cudaPointerGetAttributes(&attributes, ptr);
```

Comment:
I added them because I saw it was possible to create a host memref view over device pointers etc. Would it be better if we expose this functionality to Tripy and let tripy perform explicit copies if the device pointer mismatches the expected device type?